### PR TITLE
Update to support latest level

### LINKED ISSUE #21: Crash bug with latest level
**System Information**
- Aedes: 0.42.5
- NodeJS: 14.12.0
- OS: Ubuntu 20.04 LTS
- Arch: x86_64

**Describe the bug**
`aedes-persistence-level` crashes with latest `level`. 

**To Reproduce**

I've created a repository that reproduces the crash here: https://github.com/ralphtheninja/aedes-crash-with-level
How to reproduce is described in the readme.

**Expected behavior**
It shouldn't crash.

**Additional context**
It only seems to happen when you publish a `qos: 2` message.

### PR DESCRIPTION
**Pull request by ralphtheninja**
Fixes some problems with missing `{ valueEncoding: 'binary' }` in calls to `db.get()`. Without them, we get strings back which makes `msgpack` very unhappy.

Updated tests using latest `level` instead of `levelup` + `memdown`.

Closes https://github.com/moscajs/aedes-persistence-level/issues/21

### CODE REVIEW

**Comment by ralphtheninja** [comment]
Squash when happy.

File: persistence.js [hunk 1]
```diff
@@ -83,11 +89,7 @@ LevelPersistence.prototype.createRetainedStream = function (pattern) {
     }
   })
 
-  pump(this._db.createValueStream({
-    gt: 'retained:',
-    lt: 'retained\xff',
```
Comment (ralphtheninja) [inline persistence.js:89]: I think this was actually a 'bug'. Technically the `lt` should've been `'retained:\xff'` (with a colon), compared to all other keys, e.g. `SUBSCRIPTIONS` etc.

File: test.js [hunk 2]
```diff
@@ -3,22 +3,22 @@
...
 var abs = require('aedes-persistence/abstract')
-var levelup = require('levelup')
-var memdown = require('memdown')
+var level = require('level')
+var tempy = require('tempy')
 
-function memdb () {
-  return levelup(memdown())
+function leveldb () {
+  return level(tempy.directory())
```
Comment (ralphtheninja) [inline test.js:10]: `tempy.directory()` is very useful, we use it in `leveldown` and other modules.

Comment (robertsLando) [reply test.js:10]: :+1:

File: persistence.js [hunk 3]
```diff
@@ -21,6 +21,16 @@ var OUTGOING = 'outgoing:'
 var OUTGOINGID = 'outgoing-id:'
 var INCOMING = 'incoming:'
 var WILL = 'will:'
+var encodingOption = {
```
Comment (robertsLando) [inline persistence.js]: Could you transform all this including modules require to `const`?

**Review by robertsLando — CHANGES_REQUESTED** [review]

File: persistence.js (hunk 1 above)
Comment (robertsLando) [reply persistence.js:89]: I think it is! :+1:

File: persistence.js (hunk 3 above)
Comment (ralphtheninja) [reply persistence.js]: Sure, no problem. I was thinking about this as well.

File: persistence.js [hunk 4]
```diff
@@ -145,9 +145,9 @@ LevelPersistence.prototype.addSubscriptions = function (client, subs, cb) {
   subs = subs
     .map(withClientId, client)
     .map(addSubToTrie, this._trie)
-  let opArray = []
+  const opArray = []
```
Comment (ralphtheninja) [inline persistence.js]: `standard` complains about this not being reassigned so `let` -> `const`

File: persistence.js [hunk 5]
```diff
@@ -247,7 +247,7 @@ LevelPersistence.prototype.subscriptionsByTopic = function (pattern, cb) {
 }
 
 LevelPersistence.prototype.cleanSubscriptions = function (client, cb) {
-  var that = this
+  const that = this
```
Comment (ralphtheninja) [inline persistence.js:251]: What's your thoughts on using arrow functions to get rid of `that` variables?

File: persistence.js [hunk 6]
```diff
@@ -260,16 +260,16 @@ LevelPersistence.prototype.cleanSubscriptions = function (client, cb) {
...
 }
 
 LevelPersistence.prototype.outgoingEnqueueCombi = function (subs, packet, cb) {
   if (!subs || subs.length === 0) {
     return cb(null, packet)
   }
-  var count = 0
-  for (var i = 0; i < subs.length; i++) {
+  let count = 0
+  for (let i = 0; i < subs.length; i++) {
```
Comment (ralphtheninja) [inline persistence.js:273]: I know you can use `const` in for-loops, but I think it looks weird.

File: persistence.js [hunk 7]
```diff
@@ -393,12 +393,12 @@ LevelPersistence.prototype.incomingGetPacket = function (client, packet, cb) {
 }
 
 LevelPersistence.prototype.incomingDelPacket = function (client, packet, cb) {
-  var key = INCOMING + client.id + ':' + packet.messageId
+  const key = INCOMING + client.id + ':' + packet.messageId
```
Comment (ralphtheninja) [inline persistence.js]: We should really use `${INCOMING}${clientId}:${package.messageId}` imo

Comment (robertsLando) [reply persistence.js]: Go for the template syntax yes!

**Review by robertsLando — APPROVED** [review]
lgtm

**Comment by ralphtheninja** [comment]
Not running github actions on PRs?

**Comment by robertsLando** [comment]
> Not running github actions on PRs?

Github bug? Cannot see nothing on actions tab :confused:

**Comment by robertsLando** [comment]
Oh totally missed that :fearful: You have to place github workflow in `.github/workflows/` folder

**Comment by robertsLando** [comment]
@ralphtheninja I have made a pr to fix that #25

File: persistence.js (hunk 5 above)
Comment (robertsLando) [reply persistence.js:251]: @ralphtheninja I prefer to keep `that`, also in all other projects we use this style :)

**Comment by robertsLando** [comment]
Let's wait for #25 and #26 to land then pull master and once ready this can be merged too

**Comment by ralphtheninja** [comment]
> Oh totally missed that fearful You have to place github workflow in `.github/workflows/` folder

Sorry. I knew this, but screwed up anyway :)

**Comment by robertsLando** [comment]
@mcollina last ping request (for today) :)

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by robertsLando** [comment]
@mcollina please add me as a collaborator of this in npm and/or move the package under the moscajs org in order to be able to publish a new major release 

https://www.npmjs.com/package/aedes-persistence-level

**Pull request merged**